lib/puppet/provider/zone/solaris.rb

```diff
@@ -263,7 +263,8 @@ def start
 
       unless Puppet::FileSystem.exist?(sysidcfg)
         begin
-          File.open(sysidcfg, "w", 0600) do |f|
+          # For compatibility reasons use System encoding for this OS file
+          Puppet::FileSystem.open(sysidcfg, "w:#{Encoding.default_external.name}", 0600) do |f|
```

Comment:
On second thought `Encoding::find('filesystem')` is used for strings in file names / paths **not** file contents per https://docs.ruby-lang.org/en/2.1.0/Encoding.html#class-Encoding-label-Filesystem+encoding

So this either needs to change to `utf-8` or remain as `Encoding.default_external.name` (since we're passing in a string)